march_shared_classes/src/march_shared_classes/gait/joint_trajectory.py

```diff
@@ -88,9 +88,12 @@ def interpolate_setpoints(self):
         for i in range(0, len(time)):
             yi.append([position[i], velocity[i]])
 
-        bpoly = BPoly.from_derivatives(time, yi)
+        # We do a cubic spline here, just like the ros joint_trajectory_action_controller,
+        # see https://wiki.ros.org/robot_mechanism_controllers/JointTrajectoryActionController
+        position = BPoly.from_derivatives(time, yi)
+        derivative = position.derivative()
```

Comment:
Derivative of position is velocity, right?
```suggestion
        velocity = position.derivative()
```